virtual/environment.go

```diff
@@ -726,10 +762,13 @@ func (r *environment) maybeLogHeartbeatState(
 	numActors int,
 	usedMemory int,
 ) {
+	r.heartbeatState.Lock()
```

Comment:
Can you leave a comment here like: "Normally this synchronization is not required because this function is called in a single-threaded loop, but we have additional synchronization here because some integration tests will force heartbeats to occur manually"